Neos.Flow/Classes/Http/Middleware/MiddlewaresChainFactory.php

```diff
@@ -0,0 +1,84 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Http\Middleware;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Http\Component\ComponentContext;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Utility\PositionalArraySorter;
+use Psr\Http\Server\MiddlewareInterface;
+
+/**
+ * Creates a new MiddlewaresChain according to the specified settings
+ *
+ * @Flow\Scope("singleton")
+ */
+class MiddlewaresChainFactory
+{
+    /**
+     * @Flow\Inject
+     * @var ObjectManagerInterface
+     */
+    protected $objectManager;
+
+    /**
+     * @var ComponentContext
+     */
+    protected $componentContext;
+
+    /**
+     * MiddlewaresChainFactory constructor.
+     * @param ComponentContext $componentContext
+     */
+    public function __construct(ComponentContext $componentContext)
+    {
+        $this->componentContext = $componentContext;
+    }
+
+    /**
+     * @param array $chainConfiguration
+     * @param string $parentChain
+     * @return MiddlewaresChain
+     * @throws Exception
+     */
+    public function create(array $chainConfiguration, $parentChain = 'default'): MiddlewaresChain
+    {
+        if (empty($chainConfiguration)) {
+            return new MiddlewaresChain($parentChain);
+        }
+        $arraySorter = new PositionalArraySorter($chainConfiguration);
+        $sortedChainConfiguration = $arraySorter->toArray();
+
+        $middlewaresChain = [];
+        foreach ($sortedChainConfiguration as $middlewareName => $configuration) {
+            if (isset($configuration['chain'])) {
+                $middleware = $this->create($configuration['chain'], $middlewareName);
```

Comment:
Is the 'chain' in 'chain' still a thing or is this a leftover?

Reply:
Would be nice to make the middleware nullable. Since some might have to disable a middleware that is added by a package.

Reply:
I left the 'chain' in, because I do think there might be a use-case to configuring different sub-chains that describe a processing step.

> Would be nice to make the middleware nullable.

What about just replacing by a `NopMiddleware`? I'm not so much a fan of `'foo': null` as a measure to "disable a middleware"

Reply:
A nop or passtrough middleware would fine aswell.

Reply:
Regarding sub chains. I would like to handle this as middleware once someone needs it and maybe even in a non-core package. Offcourse this would require to pass configuration to the middleware constructor/factory from the settings. I see this as a possible feature for later.

However configuring a middleware via setting may be useful anyways as some middlewares i've seen like the "middlewares/minifier" will be put into the same chain more then once.  Not sure the Objects.yaml with virtual Objects is a good final solution for that.

So this might mean to leave the `chain` out now as it adds api we may not want to support and maybe add middleware configuration via settings as a later feature.